src/services/sentry.rs

```
@@ -46,7 +47,16 @@ impl SentryService {
         if let Some(request) = request {
             let data = request.data;
             if let Some(id) = ProtoMessageId::from_i32(request.id) {
-                let id = EthMessageId::from(id).to_usize().unwrap();
+                let eth_id = EthMessageId::try_from(id);
+                if let Err(error) = eth_id {
+                    warn!(
+                        "SentryService send_by_predicate ignores an unsupported message: {:?}",
+                        error
+                    );
+                    return SentPeers { peers };
+                }
+
+                let id = eth_id.unwrap().to_usize().unwrap();
```

if let + unwrap 🙄 

replace with match

---

I did this on purpose to not touch the lines below. This code heavily uses optionals, and this leads to an extreme nestedness (up to 10 levels). If I replace with match, it becomes 11 levels :)

Some other languages have guards or early returns. I haven't found anything like that in Rust, except using a macro like here: https://stackoverflow.com/a/51345372/1009546 , so I used the `if let` + `unwrap()`.

I think that we should refactor this method. We could refactor this method from Option to Result and use `?` shortcut so simplify the logic and not swallow the errors.

---

You can move `EthMessageId` conversion to individual handlers (e.g. `<SentryService as Sentry>::send_message_by_min_block`) and return `tonic::Status::invalid_argument` if it's invalid.

This way, `SentryService::send_by_predicate` stays infallible, and there will be cleaner separation between parsing gRPC response and the actual handler.

---

I've refactored it to use Result before I saw your comment. Could you check if this is fine?